<a href="https://colab.research.google.com/github/Gloria-L19/Possible-data-sets/blob/main/Selecting_datasets.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
# importing library
import pandas as pd

In [2]:
# loading data
filename=('/content/visceral_fat.csv')
df= pd.read_csv(filename)
df.head()

number  age (years)  ethnicity  diabetes mellitus  \
0       1           20        NaN                  0   
1       2           28        0.0                  0   
2       5           21        1.0                  0   
3       9           25        0.0                  0   
4      13           33        1.0                  0   

   mean diastolic bp (mmhg)  mean systolic bp (mmhg)  \
0                      73.5                    125.0   
1                      75.5                    112.0   
2                      70.5                    116.0   
3                      92.5                    167.0   
4                      79.0                    137.0   

   central armellini fat (mm) current gestational age  pregnancies (number)  \
0                        28.1                    12,1                   1.0   
1                        43.1                    11,5                   1.0   
2                        41.4                    12,6                   1.0   
3                        39.6                    16,2                   4.0   
4                        67.8                    11,2                   2.0   

   first fasting glucose (mg/dl)  bmi pregestational (kg/m)  \
0                            NaN                      33.20   
1                           76.0                      21.50   
2                           86.0                      21.41   
3                            NaN                      55.36   
4                           86.0                      30.85   

  gestational age at birth  type of delivery  child birth weight (g)  \
0                     41,0                 1                    3840   
1                     39,1                 0                    2810   
2                     39,5                 0                    2455   
3                     39,4                 0                    3400   
4                     40,2                 1                    3930   

   gestational dm  
0               0  
1               0  
2               0  
3               0  
4               1

This data was gotten from (https://www.kaggle.com/datasets/kamyababedi/visceral-adipose). This database is about predicting the gestational diabetes mellitus (GDM) using measurements of visceral adipose tissue(VAT) and other variables like age, body mass index(BMI),type of delivery,child birth weight, number of pregnancies and others as shown in the table

The target in this case is gestational dm (GDM) and it is a classification problem.

In [4]:
# Finding the number of rows and columns
df.shape

(133, 15)

The features are 14 columns

There are 133 rows in the dataset

While cleaning, i have to handle the missing data. i also have to choose the most appropriate method for predicting the data

DATASET 2

In [6]:
# loading the data
data=('/content/garments_worker_productivity.csv')
df1= pd.read_csv(data)
df1.head()

date   quarter  department       day  team  targeted_productivity  \
0  1/1/2015  Quarter1      sweing  Thursday     8                   0.80   
1  1/1/2015  Quarter1  finishing   Thursday     1                   0.75   
2  1/1/2015  Quarter1      sweing  Thursday    11                   0.80   
3  1/1/2015  Quarter1      sweing  Thursday    12                   0.80   
4  1/1/2015  Quarter1      sweing  Thursday     6                   0.80   

     smv     wip  over_time  incentive  idle_time  idle_men  \
0  26.16  1108.0       7080         98        0.0         0   
1   3.94     NaN        960          0        0.0         0   
2  11.41   968.0       3660         50        0.0         0   
3  11.41   968.0       3660         50        0.0         0   
4  25.90  1170.0       1920         50        0.0         0   

   no_of_style_change  no_of_workers  actual_productivity  
0                   0           59.0             0.940725  
1                   0            8.0             0.886500  
2                   0           30.5             0.800570  
3                   0           30.5             0.800570  
4                   0           56.0             0.800382

The data was gotten from the link below (https://www.kaggle.com/datasets/ishadss/productivity-prediction-of-garment-employees) This dataset includes important attributes of the garment manufacturing process and the productivity of the employees which had been collected manually and also been validated by the industry experts. I will be using the data to predict the actual productivity of the workers

The target is actual_productivity

This is a Regression problem.

In [7]:
# number of rows and columns in dataset 2
df1.shape

(1197, 15)

The features are  14  and rows are 1197

One of my problems would be how to hande the date, missing data but also the data consists of numerical and categorical data.